src/network.rs

```diff
@@ -57,7 +57,7 @@ impl Drop for HandleData {
     // dereferencing it if need be so the resource is freed
     fn drop(&mut self) {
         if let Some(header) = self.hptr {
-            let _ = unsafe { CString::from_raw(header.as_ptr().cast::<i8>()) };
```

Comment:
So just to reiterate what's happening - c_char is defined differently depending on your platform. See the multiple definitions in [`std::os`](https://web.mit.edu/rust-lang_v1.25/arch/amd64_ubuntu1404/share/doc/rust/html/src/std/os/raw/mod.rs.html#34).

If we want it to work cross platform, we should be using the type alias (`c_char`) rather than the resolved type (i8 or u8). Or omit it altogether as you have here, using type inference, which also seems fine.

I can easily see how we got here...

[The source](https://doc.rust-lang.org/src/alloc/ffi/c_str.rs.html#397) uses the c_char type:
<img width="690" alt="Screenshot 2023-06-26 at 11 30 40" src="https://github.com/georust/proj/assets/217057/37d25d8c-a5d4-4375-8691-4c9723f1ad91">


Whereas [the rendered docs](https://doc.rust-lang.org/std/ffi/struct.CString.html#method.from_raw) use the resolved type alias:
<img width="1006" alt="Screenshot 2023-06-26 at 11 31 29" src="https://github.com/georust/proj/assets/217057/20e8bd34-c88f-4ab8-9bb7-89777d90fe3f">

Presumably that's dependent on whatever arch the docs were targeting, so it will be inaccurate for people on those other platforms, and fail to compile. 

I'm not sure why that is, and I don't have time to investigate further now, but maybe it's: https://github.com/rust-lang/rust/issues/15823

Reply:
Ahhh I saw that. `c_char` and was confused by it, because I treated the docs as canonical, like the fool that I am. Good catch.